release-notes/cabal-install-3.10.1.0.md

```diff
@@ -0,0 +1,227 @@
+
+  - Targets of the `run` command are pretty-printed when failing due to multiple targets.
+  - Duplicate targets are removed in the output.
+
+- Expose BuildFlags and ClientHaddockFlags [#8351](https://github.com/haskell/cabal/issues/8351) [#8353](https://github.com/haskell/cabal/pull/8353)
+
+  Some types are present in public functions in cabal-install, but they are not exported.
+  This PR fix it for BuildFlags and ClientHaddockFlags.
+
+- `cabal man` handles $PAGER containing arguments now [#8405](https://github.com/haskell/cabal/issues/8405) [#8353](https://github.com/haskell/cabal/pull/8353)
+
+  Things like `PAGER="less -FX" cabal man` work now.
+
+  There's a slight change in the default behavior. We still use `less -R` as the default,
+  but if the user passes `PAGER=less`, we won't add -R to that, unlike before. This is
+  reasonable, as the user should be able to set the variable as they see fit.
+
+- Fix issue with "nix" config option [#8452](https://github.com/haskell/cabal/issues/8452) [#8522](https://github.com/haskell/cabal/pull/8522)
+
+  Nix option in config file was broken with #8054, this should fix it.
+
+- "Build profile" message now reflects optimization level set in global config [#8487](https://github.com/haskell/cabal/issues/8487) [#8488](https://github.com/haskell/cabal/pull/8488)
+
+  Imagine you have `optimization: 2` in your `~/.cabal/config`, and you call `cabal build`
+  in a project that doesn't have optimization level explicitly set in its project file.
+  You will still see 'Build profile: -w ghc-<VER> -O1'. This is incorrect and was fixed
+  in this patch: now you'll see '-O2'.
+
+- `ghc`, `ghc-boot`, `ghci` packages should be non-upgradable [#8489](https://github.com/haskell/cabal/issues/8489) [#8501](https://github.com/haskell/cabal/pull/8501)
+
+  - Changed both `Distribution.Client.Dependency`'s `dontUpgradeNonUpgradeablePackages`
+    and `Distribution.Solver.Modular.Solver`'s `nonInstallable` to be in sync.
+
+- Apply command line flags to install packages [#8637](https://github.com/haskell/cabal/issues/8637) [#8779](https://github.com/haskell/cabal/pull/8779)
+
+  - Command line flags usually only apply to "local" packages (packages specified
+    in the cabal.project). This change causes the v2-install command to ignore
+    that distinction to better match the expected behavior for packages specified
+    directly in the command.
+
+- Make sure Haskell files in explicit source directories take precedence over autogenerated Haskell files [#8689](https://github.com/haskell/cabal/issues/8689) [#8690](https://github.com/haskell/cabal/pull/8690)
+
+  - Changed order or directories in GHC invocation so that source
+    directories explicitly specified in cabal file will be considered
+    before Cabal’s internal build directory.
+
+- Added haddock-project command [#7669](https://github.com/haskell/cabal/issues/7669) [#8162](https://github.com/haskell/cabal/pull/8162)
```

Comment:
Adding a new subcommand sounds like a significant change to me. Pull? Also, backticks:

```suggestion
- Added `haddock-project` command [#7669](https://github.com/haskell/cabal/issues/7669) [#8162](https://github.com/haskell/cabal/pull/8162)
```